### PR TITLE
feat: html escape docstrings

### PR DESCRIPTION
**Pull request by alexjbest**
also fix the escape codes

see https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/ConditionallyCompleteLattice/Basic.html#ciSup_le for examples of why this is needed

### CODE REVIEW

File: DocGen4/Output/DocString.lean [hunk 1]
```diff
@@ -210,11 +210,11 @@ partial def modifyElement (element : Element) : HtmlM Element :=
 
 /-- Convert docstring to Html. -/
 def docStringToHtml (s : String) : HtmlM (Array Html) := do
-  let rendered := CMark.renderHtml s
+  let rendered := CMark.renderHtml (Html.escape s)
   match manyDocument rendered.mkIterator with
   | Parsec.ParseResult.success _ res =>
     res.mapM fun x => do return Html.text <| toString (← modifyElement x)
-  | _ => return #[Html.text rendered]
+  | _ => return #[Html.text <| rendered]
```
Comment (alexjbest) [inline DocGen4/Output/DocString.lean:217]:
```suggestion
  | _ => return #[Html.text rendered]
```

**Review by hargoniX — APPROVED** [review]
Thanks!

**Comment by acmepjz** [comment]
Unfortunately this PR breaks the `<url>` parsing in markdown.

Could you explain why this was needed (as of Oct. 2023)? Seems that currently it's not needed anymore in your example. I checked the git history of that file in Oct. 2023, and have no idea why it was needed at that time (perhaps because old version of doc-gen4 throws the declaration itself to markdown processor?)

**Comment by alexjbest** [comment]
Check out
https://web.archive.org/web/20230325075237/https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/ConditionallyCompleteLattice/Basic.html#ciSup_le
and scroll around a bit. Some parts of the page have completely broken
formatting.

**Comment by acmepjz** [comment]
> Some parts of the page have completely broken formatting.

I see. Seems that some code blocks are randomly duplicated. But I'm not sure this is related to the unescape thing. On the other hand, we come up with another solution <https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Contribution.20to.20doc-gen4.3F/near/443537944>. It turns out that [Lean.Xml.instToStringElement](https://leanprover-community.github.io/mathlib4_docs/find/?pattern=Lean.Xml.instToStringElement#doc) is not appropriate here (or perhaps it's buggy <https://github.com/leanprover/lean4/issues/4411>). Could you have a loot on it?

**Comment by alexjbest** [comment]
Unfortunately I don't have time to look at the moment. But if you have an implementation that you've tested works fine on this page and others that have `<` characters in it's probably fine.

**Pull request merged**